lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -169,7 +172,11 @@ impl ShardOperation for LocalShard {
             .await
     }
 
-    async fn count(&self, request: Arc<CountRequestInternal>) -> CollectionResult<CountResult> {
+    async fn count(
+        &self,
+        request: Arc<CountRequestInternal>,
+        _timeout: Option<Duration>,
+    ) -> CollectionResult<CountResult> {
         let total_count = if request.exact {
             let all_points = self.read_filtered(request.filter.as_ref())?;
```

Comment:
I think it makes sense to enable cancellation of `read_filtered`

Reply:
It actually requires a bit more work to do it properly, I propose to do this in a subsequent PR.

Reply:
https://github.com/qdrant/qdrant/pull/4842/commits/81b42890a14d5849f7c137b41a59f47de10f9c0d

Reply:
Ok